hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OFSPath.java

```diff
@@ -263,6 +264,23 @@ public boolean isBucket() {
         !this.getVolumeName().isEmpty();
   }
 
+  /**
+   * If volume and bucket names are not empty and the key name
+   * only contains the snapshot indicator, then return true.
+   * e.g. /vol/bucket/.snapshot is a snapshot indicator.
+   */
+  public boolean isSnapshotIndicator() {
+    if (keyName.contains(OM_SNAPSHOT_INDICATOR)) {
```

Comment:
I feel like we should check for key prefix (`OM_KEY_PREFIX + OM_SNAPSHOT_INDICATOR` ?) rather than `OM_SNAPSHOT_INDICATOR`.

Because we are not preventing users from creating a regular key with `.snapshot` in its name?

Reply:
@smengcl I get what you are saying, you are right! Key name might be `sn.snapshot` and this method would still accept it. I'll fix it!

Reply:
Just curious if we should prevent user from creating key with `.snapshot` in it because it is kind of reserved keyword for Ozone.

Reply:
there is a ticket for that here: https://issues.apache.org/jira/browse/HDDS-7168

Reply:
Thanks @GeorgeJahad